apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/ActivationType.java

```diff
@@ -0,0 +1,134 @@
+    PROGRAMMATIC_SELF_ATTACH, // ElasticApmAttacher.attach();
+    JAVAAGENT_FLAG, // -javaagent:... command-line option used
+    AWS_LAMBDA_LAYER, // Only if installed by using layers (as other metadata already identifies lambda)
+    ENV_ATTACH, // JAVA_TOOL_OPTIONS env var used to specify -javaagent option
+    APM_AGENT_ATTACH_CLI, // 'java -jar apm-agent-attach-cli.jar ...' used
+    FLEET, // Fleet using 'java -jar apm-agent-attach-cli.jar ...' to attach (directly or through webhook)
+
+    AZURE_FUNCTIONS, //Not defined yet how this is identified
+    UNKNOWN;
+
+    private static final Logger logger = LogManager.getLogger(ActivationType.class);
+
+    public String toReferenceString() {
+        return toString().replace('_', '-').toLowerCase();
+    }
+
+    public static ActivationType findType() {
+        ActivationType type = _findType();
+        logger.debug("findType() returning {}",type);
+        return type;
+    }
+
+    public static ActivationType _findType() {
+        // Really just a big `switch` type method
+
+        ElasticApmTracer tracer = GlobalTracer.getTracerImpl();
+        if (tracer == null) {
+            //Not yet finished initializing, or we're in test mode, in which case
+            return UNKNOWN;
+        }
+
+        CoreConfiguration.ActivationMethod method = tracer.getConfig(CoreConfiguration.class).getActivationMethod();
+        logger.debug("CoreConfiguration.ActivationMethod: {}", method);
+        if (!method.equals(CoreConfiguration.ActivationMethod.NONE)){
+            //the activation method was specified by the attacher, so use this as best option
+            switch (method) {
+                case K8S: return K8S_ATTACH;
+                case REMOTE: return APM_AGENT_ATTACH_CLI;
+                case SELF_ATTACH: return PROGRAMMATIC_SELF_ATTACH;
+                case FLEET: return FLEET;
+            }
+        }
+
+        //The next picks up `Container image' lambdas but only if the layer is used, which we
+        //aren't sure is good or bad. We could also check ELASTIC_APM_AWS_LAMBDA_HANDLER
+        //but have deliberately decided not to (for now)
+        String lambdaWrapper = PrivilegedActionUtils.getEnv("AWS_LAMBDA_EXEC_WRAPPER");
+        logger.debug("AWS_LAMBDA_EXEC_WRAPPER: {}", lambdaWrapper);
+        if (null != lambdaWrapper && lambdaWrapper.equals("/opt/elastic-apm-handler")) {
+            return AWS_LAMBDA_LAYER;
+        }
+
+        String elasticJavaagentOnTheCommandline = getElasticJavaagentOnTheCommandline();
+        logger.debug("Agent on command line: {}", elasticJavaagentOnTheCommandline);
+        String javaToolOptions = PrivilegedActionUtils.getEnv("JAVA_TOOL_OPTIONS");
+        logger.debug("JAVA_TOOL_OPTIONS: {}", javaToolOptions);
+        if (null != javaToolOptions) {
+            // fallback k8s test, this is the pattern used by the webhook, but it's
+            // a fragile check as the webhook could easily be changed
```

Comment:
Similar to AWS lambda - since there is a configuration option and since we own the webhook, instead of such heuristics we can simply set this configuration option through the webhook. We already [set our own env variables](https://github.com/elastic/apm-mutating-webhook/blob/main/patch.go#L112-L165), so IIANM, we should only set `ELASTIC_APM_ACTIVATION_METHOD= K8S_ATTACH` to get this very deterministic.

Reply:
This is actually already set in the command line for the webhook, I just left this fallback because I had this before we added the command line value setting (which is picked up as the first option in the switch above). I can drop this clause entirely or leave it, there is no need to make further changes to the webhook

Reply:
Ahh - [here](https://github.com/elastic/apm-mutating-webhook/blob/85b342dd0f08f4c6a82cec87479dc85c6ce0a08e/charts/apm-attacher/values.yaml#L32).
OK then.